src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -1152,6 +1152,30 @@ private static IEnumerable<TSource> ConcatIterator<TSource>(IEnumerable<TSource>
             foreach (TSource element in second) yield return element;
         }
 
+        public static IEnumerable<TSource> Append<TSource>(this IEnumerable<TSource> source, TSource element)
+        {
+            if (source == null) throw Error.ArgumentNull("source");
+            return AppendIterator<TSource>(source, element);
+        }
+
+        private static IEnumerable<TSource> AppendIterator<TSource>(IEnumerable<TSource> source, TSource element)
+        {
+            foreach (TSource e1 in source) yield return e1;
+            yield return element;
+        }
+
+        public static IEnumerable<TSource> Prepend<TSource>(this IEnumerable<TSource> source, TSource element)
+        {
+            if (source == null) throw Error.ArgumentNull("source");
+            return PrependIterator<TSource>(source, element);
+        }
+
+        private static IEnumerable<TSource> PrependIterator<TSource>(IEnumerable<TSource> source, TSource element)
+        {
+            yield return element;
+            foreach (TSource e1 in source) yield return e1;
```

Comment:
because element is already taken by the parameter :-)
I saw other methods use e1, e2 ... in such situations.